lib/src/main/scala/spinal/lib/sim/GenerateDiagram.scala

```diff
@@ -0,0 +1,510 @@
+    }
+    builder ++= s"""]\n}\n"""
+  }
+/** Generating HDElk language for Clock legend*/
+  def beginDraw():Unit = {
+    builder ++=
+      s"""
+         |<div id="${topNode.labelName}"></div>
+         |<h3>${topNode.labelName}</h3><br><br><br><br>
+         |<script type="text/javascript">
+         |
+         |var mygraph = {
+         |children:[
+         |""".stripMargin
+    dealAllSignals()
+    drawNodes(topNode)
+    if (clkMap.nonEmpty)
+      drawClockDomains()
+    builder ++= s"""],\n}\nhdelk.layout( mygraph,"${topNode.labelName}");\n</script>\n"""
+    val fileName = topLevelName + ".html"
+    val file = new File(fileName)
+    val pw = new FileWriter(file, true)
+    pw.write(builder.toString())
+    pw.close()
+  }
+}
+/** Integrating all methods to draw an image*/
+
+class GenerateDiagrams(rtl: SpinalReport[Component]) {
+  def beginGenerate(): Unit = {
+    val module = rtl.toplevel
+    val fileName = rtl.toplevelName + ".html"
+    val file = new File(fileName)
+    val pw = new FileWriter(file)
+    val builder = new StringBuilder()
+    builder ++=
+      s"""<!DOCTYPE html>
+         |<html>
+         |<head>
+         |    <meta charset="UTF-8">
+         |    <title>RTL连接图</title>
+         |    <style>
+         |.buttons-container {
+         | display: flex;
+         | justify-content: center;
+         | margin-top:30px;
+         |}
+         |
+         |.buttons-container button {
+         | background-color: #4CAF50;
+         | color: white;
+         | border: none;
+         | padding:10px20px;
+         | margin:010px;
+         | cursor: pointer;
+         | border-radius:5px;
+         |}
+         |
+         |.center-title {
+         |  text-align: center;
+         |  font-size:24px;
+         |  font-weight: bold;
+         |  margin-top:20px;
+         | margin-bottom: 20px;
+         |}
+         |.goTop {
+         |	width: 50px;
+         |	height: 50px;
+         |	background-color: aquamarine;
+         |	font-size: 20px;
+         |	text-align: center;
+         |	line-height: 25px;
+         |	color: azure;
+         |	position: fixed;
+         |	bottom: 50px;
+         |	right: 50px;
+         |	display: none;
+         |}
+         |    </style>
+         |</head>
+         |<body>
+         |<button class="goTop" id="gotop">Go Top</button>
+         |<script>
+         |	var goTop=document.getElementById("gotop")
+         |	    window.onscroll=function(){
+         |		var jhlheight=document.documentElement.scrollTop||document.body.scrollTop
+         |		if(jhlheight>=300){
+         |		goTop.style.display="block"
+         |		}else{
+         |		goTop.style.display="none"
+         |   }
+         |	}
+         |	  goTop.onclick=function(){
+         |		window.scrollTo({
+         |			top:0,
+         |			behavior:"smooth"
+         |		})
+         |	}
+         |</script>
+         |<script src="/js/elk.bundled.js"></script>

```

Comment:
I had to change these paths to relative ones (w/o the leading `/`), otherwise my browser would not be able to find them when opening the file locally.

Reply:
I delete the leading" / "and now it is ok.

Reply:
thats cool, should be a good default for most users - and simpler to use as well